module/VuFind/src/VuFind/ILS/Driver/Demo.php

```diff
@@ -1167,6 +1175,62 @@ public function getMyFines($patron)
         return $session->fines;
     }
 
+    /**
+     * Register a payment.
+     *
+     * This is called after a successful online payment.
+     *
+     * @param array   $patron                  Patron
+     * @param int     $amount                  Amount to be registered as paid
+     * @param string  $localPaymentIdentifier  Local payment identifier
+     * @param ?string $remotePaymentIdentifier Remote payment identifier
+     * @param int     $paymentId               Internal payment id
+     * @param ?array  $fineIds                 Fine IDs to mark paid or null for bulk payment
+     *
+     * @throws ILSException
+     * @return array Associative array with keys success (bool, always) and reason (string, on error)
+     *
+     * @SuppressWarnings(PHPMD.UnusedFormalParameter)
+     */
+    public function registerPayment(
+        array $patron,
+        int $amount,
+        string $localPaymentIdentifier,
+        ?string $remotePaymentIdentifier,
+        int $paymentId,
+        ?array $fineIds = null
+    ): array {
+        if ($this->isFailing(__METHOD__, 10)) {
+            throw new ILSException('Payment::registration_failed');
+        }
+
+        $session = $this->getSession($patron['id'] ?? null);
+        $paid = 0;
+        if (isset($session->fines)) {
+            foreach ($session->fines as $key => $fine) {
+                if (
+                    ($fine['payable_online'] ?? false)
+                    && (!$fineIds || in_array($fine['fine_id'] ?? '', $fineIds))
+                ) {
+                    unset($session->fines[$key]);
+                    $paid += $fine['balance'];
+                }
+            }
+        }
```

Comment:
Shave off some lines?

```suggestion
        foreach ($session->fines ?? [] as $key => $fine) {
            if (
                ($fine['payable_online'] ?? false)
                && (!$fineIds || in_array($fine['fine_id'] ?? '', $fineIds))
            ) {
                unset($session->fines[$key]);
                $paid += $fine['balance'];
            }
        }
```